### DISCUSSION TITLE
Bug in 1 page checkout with payment custimization

### DISCUSSION THREAD

**Discussion by Sam115200** [Shopify/function-examples#307] [Feedback: Payment Customization API]
**Setup:**

- 1 payment customization active, which hides `Cash on Delivery (COD)` only when `Standard` shipping method is selected.
-  The customer email and name fields are not empty. When they are empty then there is no error.
- The `Cash on Delivery (COD)` has to be selected and then we try to hide it. If it is not selected first and we hide it, then there is no error.

**Error:**
This is the error message,
```
Your payment details couldn’t be verified. Check your card details and try again.
```

**Video:**

https://github.com/Shopify/function-examples/assets/139220490/5509f23f-363a-4e30-9257-c4b02f83d01f

In the video, 
- `Cash on Delivery (COD)` is selected and then we choose `Standard` shipping method. Payment customization kicks in and hides `Cash on Delivery (COD)` but we also gets the error message. 
-  Then selecting a different shipping method causes `Cash on Delivery (COD)` to appear again and the error is gone.
- Now, `Cash on Delivery (COD)` is not pre-selected and selecting `Standard` shipping method does cause it to be hidden and no error is generated.
-  Reproduce the error again by selecting `Cash on Delivery (COD)` and then choosing `Standard` shipping method.

**Expected Behavior:**
There should not be any error when payment customization hides any pre-selected payment method.

This error does not happen on 3-page checkout. Any help will be much appreciated. Thanks.

**Comment by bargenson** [comment]
Bug opened [here](https://github.com/Shopify/payments-platform/issues/3739).

**Reply by Sam115200** [reply]
@bargenson Thank you, but it seems like I do not have access to it. It gives me 404 error when I try to open it.

**Comment by Sam115200** [comment]
This has been resolved. Thank you team.
